### PR TITLE
Specify external meta attributes file in GUI

### PR DESCRIPTION
**Pull request by signedav**
Specify an external meta attributes file (.toml) in Advanced Options and provide it to ili2db by `--iliMetaAttrs`

projectgen

Issue #184

Missing:
- [ ] Test
- [ ] Docs
- [ ] loading toml-file from repository

 (Plus small fix, that on Advanced Options with "Cancel" the settings are not stored)

### CODE REVIEW

File: projectgenerator/gui/generate_project.py [hunk 1]
```diff
@@ -348,6 +348,7 @@ def updated_configuration(self):
 
         configuration.epsg = self.epsg
         configuration.inheritance = self.ili2db_options.get_inheritance_type()
+        configuration.tomlfile = self.ili2db_options.get_toml_file()
```
Comment (m-kuhn) [inline projectgenerator/gui/generate_project.py]: Can you remove the `get_` prefix (we also never do that on qgis side). Also remove it on the inheritance type while on it.

File: projectgenerator/gui/ili2db_options.py [hunk 2]
```diff
@@ -18,6 +18,14 @@
  ***************************************************************************/
 """
 
+from projectgenerator.utils.qt_utils import (
+    make_file_selector,
+    make_save_file_selector,
+    Validators,
+    FileValidator,
+    NonEmptyStringValidator,
+    OverrideCursor
```
Comment (m-kuhn) [inline projectgenerator/gui/ili2db_options.py]: That's a lot of imports there, of which I think some are not used. (e.g. can't find `OverrideCursor`)
PyCharm will list them in gray

File: projectgenerator/libili2db/ili2dbconfig.py [hunk 3]
```diff
@@ -161,6 +162,9 @@ def to_ili2db_args(self, hide_password=False):
         if self.ilimodels:
             args += ['--models', self.ilimodels]
 
+        if self.tomlfile:
+            args += ["--iliMetaAttrs", self.tomlfile]
```
Comment (m-kuhn) [inline projectgenerator/libili2db/ili2dbconfig.py]: @marioba do you remember, is this parameter for all ili2db commands, or only schemaimport?

**Comment by m-kuhn** [comment]
Thanks @signedav looks really good

**Comment by m-kuhn** [comment]
sidenote: toml file from repository is not yet on the todo list, this can be skipped for now.

File: projectgenerator/gui/ili2db_options.py (hunk 2 above)
Comment (signedav) [reply projectgenerator/gui/ili2db_options.py]: Yes. I don't use `make_save_file_selector` and `OverrideCursor` - I forgot to remove them after copy this part...

**Comment by signedav** [comment]
Okay, so I will make the tests and docs and then we communicate before I would start with repo-stuff. Thanks.

**Pull request closed without merging**